### PR TITLE
Create simple makefile to simplify installation procedure

### PR DESCRIPTION
**Pull request by Prototik**
Contributions welcome

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+INSTALL_PROGRAM = install
```
Comment (ignatenkobrain) [inline Makefile:1]: I think this should be just `INSTALL ?= install`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+CARGO_PROGRAM = cargo
```
Comment (ignatenkobrain) [inline Makefile:2]: same here, just a `CARGO`

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+check: build
+	@$(CARGO_PROGRAM) test --release
+
+clean:
+	@$(CARGO_PROGRAM) clean
+
+install: build
+	$(INSTALL_PROGRAM) -Dm755 target/release/zram-generator $(DESTDIR)$(SYSTEMD_DIR)/system-generators/zram-generator
+	$(INSTALL_PROGRAM) -Dm644 units/swap-create@.service $(DESTDIR)$(SYSTEMD_DIR)/system/swap-create@.service
+	$(INSTALL_PROGRAM) -Dm644 zram-generator.conf.example $(DESTDIR)$(PREFIX)/share/doc/zram-generator/zram-generator.conf.example
```
Comment (ignatenkobrain) [inline Makefile:21]: these should have `-p` to preserve timestamps.

File: README.md [hunk 4]
```diff
@@ -24,9 +24,11 @@ This program serves two purposes:
...
-Copy the `zram-generator` binary  into `/usr/lib/systemd/system-generators/`,
-    `units/swap-create@.service`  into `/usr/lib/systemd/system/`,
-and `zram-generator.conf.example` into `/etc/systemd/zram-generator.conf`, customising it to your liking.
+Executing `make install` will create the following things:
+* Generator binary installed as `/usr/lib/systemd/system-generators/zram-generator`
+* `units/swap-create@.service` copied into `/usr/lib/systemd/system/`
+* `zram-generator.conf.example` copied into `/usr/share/doc/zram-generator/`
+You need though create your own config file at `/etc/systemd/zram-generator.conf`, customising it to your liking.
 
 The "{generator}" template in `units/swap-create@.service.d/binary-location.conf`
```
Comment (nabijaczleweli) [inline README.md:33]: This is better moved into the Testing sexion now.

**Comment by Prototik** [comment]
> How about an `uninstall` target that removes the three files?

That's not common usecase, usually `install` target used in packaging scripts and uninstalling handled by package manager, not makefiles.

**Comment by nabijaczleweli** [comment]
I've seen it around, but more than fair enough.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,21 @@
+INSTALL_PROGRAM = install
+CARGO_PROGRAM = cargo
+PREFIX = /usr
+SYSTEMD_DIR ?= $(PREFIX)/lib/systemd
```
Comment (keszybz) [inline Makefile:4]: This could also be `pkgconf systemd --variable=systemd_system_unit_dir`, but that is fine too.

**Comment by keszybz** [comment]
```
/usr/share/rust/.cargo/bin/cargo test --verbose
error: failed to download `remove_dir_all v0.5.2`
```
cargo seems to be having a bad day.

**Pull request merged**